pkg/utils/ssh/ssh.go

```diff
@@ -98,6 +98,21 @@ func NewSshExecWithKeyFile(logger *utils.Logger, user string, file string) *SshE
 	return sshexec
 }
 
+// This function requires the password string to be crypt encrypted
+func NewSshExecWithPassword(logger *utils.Logger, user string, password string) *SshExec {
```

Comment:
this deserves a separate commit (in same PR).

note that I prefer amended force-pushes over add-on patches